packages/grafana-ui/src/components/uPlot/Plot.tsx

```diff
@@ -117,12 +141,150 @@ export class UPlotChart extends React.Component<PlotProps, UPlotChartState> {
+  handlePressedKeys = (time: number) => {
+    const nothingPressed = Object.values(this.keyboardState.pressedKeys).every((pressed) => pressed === false);
+    if (nothingPressed || !this.state.plot) {
+      this.keyboardState.keysLastHandledAt = null;
+      return;
+    }
+
+    const dt = time - (this.keyboardState.keysLastHandledAt ?? time);
+    const dx = dt * PIXELS_PER_MS;
+    let horValue = 0;
+    let vertValue = 0;
+
+    if (this.keyboardState.pressedKeys.ArrowUp) {
+      vertValue -= dx;
+    }
+    if (this.keyboardState.pressedKeys.ArrowDown) {
+      vertValue += dx;
+    }
+    if (this.keyboardState.pressedKeys.ArrowLeft) {
+      horValue -= dx;
+    }
+    if (this.keyboardState.pressedKeys.ArrowRight) {
+      horValue += dx;
+    }
+    if (this.keyboardState.pressedKeys.Shift) {
+      horValue *= SHIFT_MULTIPLIER;
+      vertValue *= SHIFT_MULTIPLIER;
+    }
+
+    this.moveCursor(horValue, vertValue);
+
+    const cursor = this.state.plot.cursor;
+    if (this.keyboardState.pressedKeys[' '] && cursor) {
+      const drawHeight = Number(this.state.plot.over.style.height.slice(0, -2));
+
+      this.state.plot.setSelect(
+        {
+          left: cursor.left! < this.keyboardState.dragStartX! ? cursor.left! : this.keyboardState.dragStartX!,
+          top: 0,
+          width: Math.abs(cursor.left! - (this.keyboardState.dragStartX ?? cursor.left!)),
+          height: drawHeight,
+        },
+        false
+      );
+    }
+
+    this.keyboardState.keysLastHandledAt = time;
+    window.requestAnimationFrame(this.handlePressedKeys);
+  };
+
+  moveCursor = (dx: number, dy: number) => {
+    if (this.state.plot?.cursor.left === undefined) {
+      return;
+    }
+
+    const { cursor } = this.state.plot;
+    this.state.plot.setCursor({
+      left: cursor.left! + dx,
+      top: cursor.top! + dy,
+    });
+  };
+
+  handleFocus = () => {
+    // Is there a more idiomatic way to do this?
+    const drawWidth = Number(this.state.plot?.over.style.width.slice(0, -2));
+    const drawHeight = Number(this.state.plot?.over.style.height.slice(0, -2));
+
+    this.state.plot?.setCursor({ left: drawWidth / 2, top: drawHeight / 2 });
+  };
+
+  handleBlur = () => {
+    this.keyboardState = initialKeyboardState;
+    this.state.plot?.setSelect({ left: 0, top: 0, width: 0, height: 0 }, false);
+  };
+
   render() {
     return (
       <div style={{ position: 'relative' }}>
-        <div ref={this.plotContainer} data-testid="uplot-main-div" />
+        <ThemeContext.Consumer>
+          {(theme) => {
+            const styles = getStyles(theme);
+            return (
+              <div
+                className={styles.focusStyle}
+                ref={this.plotContainer}
+                data-testid="uplot-main-div"
+                tabIndex={0}
+                onFocusCapture={this.handleFocus}
+                onBlur={this.handleBlur}
+                onKeyDown={this.handleKeys}
+                onKeyUp={this.handleKeyRelease}
```

Comment:
naming of the callbacks breaks our current style guide, https://github.com/grafana/grafana/blob/b164ce0c9bad2b2253d5e2c32d8d574d9bfcbe36/contribute/style-guides/frontend.md#name-callback-props-and-handlers-with-an-on-prefix  
If many are not happy with the current style we could bring this up on frontend weekly and change it or allow for both